common/src/gas_provider/negative_imbalance.rs

```diff
@@ -17,111 +17,69 @@
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 use super::*;
-use core::cmp::Ordering;
 
 /// Opaque, move-only struct with private field to denote that value has been
 /// destroyed without any equal and opposite accounting.
 #[derive(RuntimeDebug, PartialEq, Eq)]
-pub struct NegativeImbalance<Balance: BalanceTrait, TotalValue: ValueStorage<Value = Balance>>(
-    Balance,
-    PhantomData<TotalValue>,
-);
-
-impl<Balance: BalanceTrait, TotalValue: ValueStorage<Value = Balance>>
-    NegativeImbalance<Balance, TotalValue>
-{
+pub struct NegativeImbalance<Balance: BalanceTrait>(Balance);
+
+impl<Balance: BalanceTrait> NegativeImbalance<Balance> {
     /// Create a new negative imbalance from value amount.
     pub fn new(amount: Balance) -> Self {
-        NegativeImbalance(amount, PhantomData)
+        NegativeImbalance(amount)
     }
-}
 
-impl<Balance: BalanceTrait, TotalValue: ValueStorage<Value = Balance>> TryDrop
-    for NegativeImbalance<Balance, TotalValue>
-{
-    fn try_drop(self) -> Result<(), Self> {
-        self.drop_zero()
+    /// Applies imbalance to some amount.
+    pub fn apply_to(&self, amount: &mut Option<Balance>) {
+        let new_value = amount.unwrap_or_else(Zero::zero);
+        if self.0 > new_value {
+            log::debug!(
+                target: "essential",
+                "Unaccounted gas detected: burnt {:?}, known total supply was {:?}.",
+                self.0,
+                *amount
+            )
+        }
+
+        *amount = Some(new_value.saturating_sub(self.0));
     }
 }
 
-impl<Balance: BalanceTrait, TotalValue: ValueStorage<Value = Balance>> Default
-    for NegativeImbalance<Balance, TotalValue>
-{
+impl<Balance: BalanceTrait> Default for NegativeImbalance<Balance> {
```

Comment:
Default and Zero are the common traits that may be implemented for anything, but for me its meaningless for imbalances by concept